test_conformance/api/test_queue_properties_queries.cpp

```diff
@@ -0,0 +1,313 @@
+        cl_uint requested_size = 0;
+        for (cl_uint i = 0; i < requested_properties.size(); i = i + 2)
+        {
+            if (requested_properties[i] == CL_QUEUE_PROPERTIES)
+            {
+                requested_bitfield = requested_properties[i + 1];
+            }
+            if (requested_properties[i] == CL_QUEUE_SIZE)
+            {
+                requested_size = requested_properties[i + 1];
+            }
+        }
+
+        error = verify_if_properties_supported(deviceID, requested_bitfield,
+                                               requested_size);
+        if (error == TEST_SKIPPED_ITSELF)
+        {
+            log_info("TC result: skipped\n");
+            return CL_SUCCESS;
+        }
+        test_error(error,
+                   "Checking which queue properties supported failed.\n");
+    }
+
+    // continue testing if supported user properties
+    error =
+        create_queue_and_check_array_properties(context, deviceID, test_case);
+    test_error(error, "create_queue_and_check_array_properties failed.\n");
+
+    log_info("TC result: passed\n");
+    return TEST_PASS;
+}
+
+int test_queue_properties_queries(cl_device_id deviceID, cl_context context,
+                                  cl_command_queue queue, int num_elements)
+{
+    int error = CL_SUCCESS;
+    std::vector<test_queue_array_properties_data> test_cases;
+
+    test_cases.push_back({ {}, "host queue, NULL properties" });
+
+    test_cases.push_back(
+        { { CL_QUEUE_PROPERTIES, 0, 0 }, "host queue, zero properties" });
+
+    test_cases.push_back(
+        { { CL_QUEUE_PROPERTIES, CL_QUEUE_PROFILING_ENABLE, 0 },
+          "host queue, CL_QUEUE_PROFILING_ENABLE" });
+
+    test_cases.push_back(
+        { { CL_QUEUE_PROPERTIES, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, 0 },
+          "host queue, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE" });
+
+    test_cases.push_back(
+        { { CL_QUEUE_PROPERTIES,
+            CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE | CL_QUEUE_PROFILING_ENABLE,
+            0 },
+          "host queue, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE | "
+          "CL_QUEUE_PROFILING_ENABLE" });
+
+    test_cases.push_back(
+        { { CL_QUEUE_PROPERTIES,
+            CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE | CL_QUEUE_ON_DEVICE, 0 },
+          "device queue, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE | "
+          "CL_QUEUE_ON_DEVICE" });
+
+    test_cases.push_back(
+        { { CL_QUEUE_PROPERTIES,
+            CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE | CL_QUEUE_ON_DEVICE
+                | CL_QUEUE_ON_DEVICE_DEFAULT | CL_QUEUE_PROFILING_ENABLE,
+            CL_QUEUE_SIZE, 124, 0 },
+          "device queue, all possible properties" });
+
+    test_cases.push_back(
+        { { CL_QUEUE_PROPERTIES,
+            CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE | CL_QUEUE_ON_DEVICE
+                | CL_QUEUE_PROFILING_ENABLE,
+            CL_QUEUE_SIZE, 124, 0 },
+          "devuce queue, all without CL_QUEUE_ON_DEVICE_DEFAULT" });
```

Comment:
```suggestion
          "device queue, all without CL_QUEUE_ON_DEVICE_DEFAULT" });
```

Reply:
Odd - my spell checker didn't catch this one.  Fixed!